src/Internal/Builtin/JsonEncoder.elm

```diff
@@ -134,14 +113,14 @@ codeGen =
                             [ fst, snd, thrd ] ->
                                 lambdaWrap "triple"
                                     (\triple ->
-                                        CG.letExpr [ CG.letDestructuring (CG.tuplePattern [ CG.varPattern "a", CG.varPattern "b", CG.varPattern "c" ]) triple ]
+                                        CG.letExpr [ CG.letDestructuring (CG.tuplePattern [ CG.varPattern "first", CG.varPattern "second", CG.varPattern "third" ]) triple ]
                                             (CG.apply
                                                 [ CG.fqFun [ "Json", "Encode" ] "list"
                                                 , CG.val "identity"
                                                 , CG.list
-                                                    [ CG.apply [ fst, CG.val "a" ]
-                                                    , CG.apply [ snd, CG.val "b" ]
-                                                    , CG.apply [ thrd, CG.val "c" ]
+                                                    [ CG.apply [ fst, CG.val "first" ]
+                                                    , CG.apply [ snd, CG.val "second" ]
+                                                    , CG.apply [ thrd, CG.val "third" ]
```

Comment:
There are some issues that sometimes pop-up with shadowing and these not being hygienic macros. Perhaps a task for another PR would be to create hygienic lambda wrappers, etc.